packages/shared-components/src/components/onto-repository-selector/onto-repository-selector-item/onto-repository-selector-item.tsx

```diff
@@ -0,0 +1,45 @@
+import {Component, Host, h, Prop} from '@stencil/core';
+import {Repository} from '@ontotext/workbench-api';
+
+@Component({
+  tag: 'onto-repository-selector-item',
+  styleUrl: 'onto-repository-selector-item.scss',
+  shadow: false,
+})
+/**
+ *
+ * A StencilJS component that renders a repository item in the repository selector dropdown. It displays the repository's
+ * icon, ID, and location. If no location is specified, it shows "Local" as the default. This component also allows for
+ * dynamic rendering of repository details as part of a list of selectable items.
+ *
+ * @example
+ * <onto-repository-selector-item repository={repository}></onto-repository-selector-item>
+ *
+ */
+export class OntoRepositorySelectorItem {
```

Comment:
This component should be FunctionalComponent

Reply:
Done